sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -31,15 +31,174 @@
 
 import operator
 import pandas
+import numpy
+import sdc
 
 from numba import types
 from numba.extending import (overload, overload_method, overload_attribute)
+from sdc.hiframes.pd_dataframe_ext import DataFrameType
+from sdc.datatypes.common_functions import TypeChecker
 from numba.errors import TypingError
+from sdc.str_arr_ext import StringArrayType
 
-from sdc.datatypes.hpat_pandas_dataframe_types import DataFrameType
+# from sdc.datatypes.hpat_pandas_dataframe_types import DataFrameType
 from sdc.utils import sdc_overload_method
 
 
+@sdc_overload_method(DataFrameType, 'append')
+def sdc_pandas_dataframe_append(df, other, ignore_index=True, verify_integrity=False, sort=None):
+    """
+    Intel Scalable Dataframe Compiler User Guide
+    ********************************************
+    Pandas API: pandas.DataFrame.append
+
+    Examples
+    --------
+    .. literalinclude:: ../../../examples/dataframe_append.py
+       :language: python
+       :lines: 27-
+       :caption: Appending rows of other to the end of caller, returning a new object.
+       Columns in other that are not in the caller are added as new columns.
+       :name: ex_dataframe_append
+    .. code-block:: console
+        > python ./dataframe_append.py
+             A  B    C
+        0  1.0  2  NaN
+        1  3.0  4  NaN
+        2  NaN  5  6.0
+        3  NaN  7  8.0
+        dtype: object
```

Comment:
I'm not sure the result will appear in the documentation due to missed empty lines around of the block.